### PR TITLE
[CIT-528] Software update response failed case

### PR DESCRIPTION
**Pull request by rina23q**
Fixed the regression after #393. Now it failed to send software list when the software update response is failed. Integration test revealed it…

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by rina23q** [comment]
Anyway, this publishing list will be replaced http request by @makr11st. It's a kind of short time fix.

File: mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs [hunk 1]
```diff
@@ -105,12 +107,23 @@ impl CumulocitySoftwareManagement {
...
+    ) -> Result<(), SMCumulocityMapperError> {
+        let response = SoftwareListResponse::from_json(json_response)?;
+
         match response.status() {
             SoftwareOperationStatus::Successful => {
-                let topic = OutgoingTopic::SmartRestResponse.to_topic()?;
-                let smartrest_response =
-                    SmartRestSetSoftwareList::from_thin_edge_json(response).to_smartrest()?;
-                let () = self.publish(&topic, smartrest_response).await?;
+                self.publish_software_list(json_response).await?;
```
Comment (makr11st) [inline mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs:126]: Is it intended to only send the list when successful and do nothing on Failed?

Comment (rina23q) [reply mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs:126]: Yes, this function is called only for `tedge/commands/res/software/list` topic. It's not an operation for c8y. Mapper should discard failed or executing for software list, unlike software update.

**Review by makr11st — APPROVED** [review]
Let's have this merged to fix the Failed.

**Pull request merged**